wallet/walletdb/interface.go

```diff
@@ -162,7 +169,9 @@ type ReadWriteCursor interface {
 // BucketIsEmpty returns whether the bucket is empty, that is, whether there are
 // no key/value pairs or nested buckets.
 func BucketIsEmpty(bucket ReadBucket) bool {
-	k, v := bucket.ReadCursor().First()
+	cursor := bucket.ReadCursor()
+	defer cursor.Close()
```

Comment:
The defer can be avoided in this simple case without introducing additional complexity:

```
cursor := bucket.ReadCursor()
k, v := cursor.First()
cursor.Close()
return k == nil && v == nil
```